src/update.lua

```diff
@@ -0,0 +1,54 @@
+--[[
+    This part of the plugin is used to fix possible errors when users update the Plugin or install it for the first time.
+]]
+
+local Plugin : Plugin
+local stringify = require(script.Parent.stringify)
+
+local currentVersion = '0.0.0' -- the current version of the plugin.
+
+local updateFunctions = {
+    none = function()
+        -- Fresh install / before 0.0.0
+        local uid = Plugin:GetSetting('user-id')
+        local uid2 = Plugin:GetSetting('UserId')
+        local tok = Plugin:GetSetting('refresh-token')
+        local ses = Plugin:GetSetting('session')
+        
+        local baseTable = {}
+        if uid or uid2 then
+            baseTable['user-id'] = uid or uid2
+            Plugin:SetSetting('user-id', nil)
+            Plugin:SetSetting('UserId', nil)
+        end
+        if tok then
+            baseTable['refresh-token'] = tok
+            Plugin:SetSetting('refresh-token', nil)
+        end
+        if ses then
+            baseTable['session'] = ses
+            Plugin:SetSetting('session', nil)
+        end
+        if baseTable ~= {} then
+            Plugin:SetSetting('auth', '!' .. stringify(baseTable))
+        else
+            Plugin:SetSetting('auth', '!{}')
+        end
+    end
+}
+
+local function update(pl)
+    Plugin = pl
+    local lastPluginVersion = Plugin:GetSetting('pluginVersion')
+    if lastPluginVersion ~= '#' .. currentVersion then
+        if lastPluginVersion == nil then
+            updateFunctions.none()
+        else
+            updateFunctions[lastPluginVersion]()
```

Comment:
This will work for now, but the last plugin version may not have needed an update. In addition, there may have been many updates since the last plugin version that all need to be run in order.

Reply:
So even if the current version doesn't have an update, there may be later versions that do need updates still.